### PR TITLE
Structured{De,}Serialize of WebAssembly.Memory

### PR DESCRIPTION
**Pull request by binji**

### CODE REVIEW

File: proposals/threads/Overview.md [hunk 1]
```diff
@@ -333,6 +333,25 @@ For the web embedding, `wake` is equivalent in behavior to executing the followi
 
 ## [JavaScript API][] changes
 
+### Structured Serialization and Deserialization of `WebAssembly.Memory`
```
Comment (rossberg-chromium) [inline proposals/threads/Overview.md]: Nit: Strictly speaking, these are changes to the Web API, not the JS API.

**Comment by domenic** [comment]
> What does it mean to serialize a shared Memory? Especially as things are being written to it.

HTML is pretty clear on this. The serialization record just contains a pointer to the shared memory. This pointer is then passed across agent boundaries, etc.

There are separate specs, like IndexedDB, which define the process of snapshotting a serialization record into bytes on disk. (Or rather, at this time, I don't believe it's well-defined, but when the spec editors have more bandwidth, they plan to do so.)

**Comment by binji** [comment]
Ah, I see. I was a bit concerned that this proposal didn't mention it anywhere, so I wanted to put something in. I guess I can just wait for that and reference it directly.

**Comment by lukewagner** [comment]
Coincidentally just talking to @lars-t-hansen about this today: if you try to send a non-`shared` `Memory`, could that perhaps just throw instead of making a copy?  B/c of the heap-mapping tricks, this may end up having surprising performance effects.

**Comment by binji** [comment]
Closing in favor of https://github.com/WebAssembly/design/pull/1074.

**Pull request closed without merging**